### PR TITLE
Changed code of conduct and updated file sync to all repos.

### LINKED ISSUE #620: Update CODE_OF_CONDUCT.md to link to new conda CoC
### What is the idea?

Now that the conda org has an official [Code of Conduct](https://github.com/conda-incubator/governance/blob/main/CODE_OF_CONDUCT.md), the [CODE_OF_CONDUCT.md](https://github.com/conda/infra/blob/6f8fb087337dbdf62abcab33267bd20415ad10a4/CODE_OF_CONDUCT.md) file should now link to that instead of NumFOCUS's CoC.

### Tasks

- [ ] ~Update local [`CODE_OF_CONDUCT.md`](https://github.com/conda/infra/blob/main/CODE_OF_CONDUCT.md) to link to conda's governance [`CODE_OF_CONDUCT.md`](https://github.com/conda-incubator/governance/blob/main/CODE_OF_CONDUCT.md)~
- [x] Sync updated `CODE_OF_CONDUCT.md` to all conda repos
- [ ] ~Remove all synced `CODE_OF_CONDUCT.md` from conda org repos~
- [ ] ~Add new CODE_OF_CONDUCT.md to https://github.com/conda/.github~
    - https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file#supported-file-types

### PR DESCRIPTION
**Pull request by kathatherine**
### Description

Closes #620. Updated the text of the current CODE_OF_CONDUCT.md to match that of the one in conda-incubator/governance. Are we going to have redundancy problems with this?

Also updated the sync config.yml so that the CODE_OF_CONDUCT.md will sync with every repo in the conda org. I didn't include the two archived ones.

### CODE REVIEW

File: .github/sync/config.yml [hunk 1]
```diff
@@ -98,3 +95,30 @@ group:
       # release docs
       - source: .github/sync/RELEASE.md
         dest: RELEASE.md
+group:
+# Code of Conduct sync
+  - repos: |
```
Comment (kenodegard) [inline .github/sync/config.yml:100]: We just need to append it to the existing group:

```suggestion
  # Code of Conduct sync
  - repos: |
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,3 +1,340 @@
-# Code of Conduct
+# Conda Organization Code of Conduct
```
Comment (kenodegard) [inline CODE_OF_CONDUCT.md:1]: Hm, do we actually want this replicated verbatim? Can we just include the short version and include a link to the full version?

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (kathatherine) [reply CODE_OF_CONDUCT.md:1]: A fair point. I was concerned about redundancy issues. That's a better solution.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
```
Comment (kenodegard) [inline CODE_OF_CONDUCT.md:2]: Thoughts on moving the link to the top and into an admonition?

```suggestion

> **Note**
> Below is the short version of our CoC, see the long version [here](https://github.com/conda-incubator/governance/blob/main/CODE_OF_CONDUCT.md).

```

Comment (kathatherine) [reply CODE_OF_CONDUCT.md:2]: Good idea.

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -1,3 +1,24 @@
...
+
+## Report an Incident
+
+* Report a code of conduct incident [using a form](https://form.jotform.com/221527028480048).
+* Report a code of conduct incident via email: [conduct@conda.org](mailto:conduct@conda.org).
+* Contact [an individual committee member](#committee-membership) or [CoC event representative](#coc-representatives) to report an incident in confidence.
+
+&nbsp;
+
+See the longer version of the Code of Conduct [here](https://github.com/conda-incubator/governance/blob/main/CODE_OF_CONDUCT.md).
```
Comment (kathatherine) [inline CODE_OF_CONDUCT.md:24]: @kenodegard Thoughts on still keeping this at the bottom? Probably redundant?

Comment (kenodegard) [reply CODE_OF_CONDUCT.md:24]: Yeah no need to keep this at the bottom

Comment (kenodegard) [inline CODE_OF_CONDUCT.md:24]:
```suggestion
```

**Review by kenodegard — APPROVED** [review]

**Pull request merged**